zenodo/modules/records/admin.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Zenodo.
+# Copyright (C) 2015, 2016 CERN.
+#
+# Zenodo is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Zenodo is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Zenodo; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Admin view for updating datacite metadata."""
+
+from datetime import datetime
+import uuid
+
+from flask import current_app, flash, redirect, request, url_for
+from flask_admin import BaseView, expose
+from flask_babelex import lazy_gettext as _
+from flask_wtf import FlaskForm
+from invenio_cache import current_cache
+from wtforms.fields.html5 import DateField
+from wtforms.validators import DataRequired
+
+from zenodo.config import DATACITE_UPDATING_RATE_PER_HOUR
+from zenodo.modules.records.utils import find_registered_doi_pids
+
+
+class UpdateDataciteForm(FlaskForm):
+    """Form for updating datacite metadata."""
+
+    field_sets = ['from_date', 'until_date']
```

Comment:
These are not really needed. The reason they exist everywhere is because of [`invenio-communities`](https://github.com/inveniosoftware/invenio-communities/blob/master/invenio_communities/forms.py#L40), which had a pretty large form and is dynamically rendered in Jinja.

For small forms it's okay to manually render fields like you've done